components/sup/src/config.rs

```diff
@@ -55,12 +59,30 @@ impl GossipListenAddr {
     }
 }
 
+// Returns a GossipListen listener address value if set in the environment. Does *not*
+// return any default value if the value was not found in the environment!
+pub fn listen_gossip_from_env() -> Option<String> {
```

Comment:
Same comments as for `listen_ctl_from_env`.

What might be useful is a general-purpose function like `socket_addr_from_env(&str: env_var, default: SocketAddr) -> SocketAddr`, since we're using that pattern several places.